training/SYMBOL_TABLE_OVERVIEW.md

```diff
@@ -0,0 +1,170 @@
+
+For more information regarding the creation of new Symbol Table Entry types, view the [Symbol Table Entry Creation](https://github.com/Bears-R-Us/arkouda/blob/master/training/SYMBOL_TABLE_ENTRY_CREATION.md) document.
+
+<a id="toc"></a>
+## Table of Contents
+1. [Symbol Table Overview](#overview)
+2. [Symbol Table Structure](#symStruct)
+3. [Creating a New Entry From an Array](#addSym)
+4. [Finding an Entry in the Symbol Table](#findSym)
+5. [Accessing Symbol Table Entry Data](#accessSym)
+
+<a id="overview"></a>
+## Symbol Table Overview
+
+The Symbol Table is implemented using a Chapel Map and serves as a `lookup table` that uses a Chapel `Map` which allows Arkouda to persist objects until the server is shutdown or the object is deleted.
+
+
+<a id="symStruct"></a>
+## Symbol Table Structure
+
+Records stored in the Symbol Table are called `Entries`. The Entry types are designed in a simple hierarchical structure as follows:
+
+- `AbstractSymEntry`
+  - `GenSymEntry`
+    - `SymEntry`
+    - `SegArraySymEntry`
+    - `SegStringSymEntry`
+  - `CompositeSymEntry`
+    - `GroupBySymEntry`
+  
+The root-level entry type is `AbstractSymEntry`. All other entry types inherit from `AbstractSymEntry` directly or from one of its children. `AbstractSymEntry` does not store any data, only metadata information, such as name and entry type.
+
+`GenSymEntry` is the first type directly inheriting from `AbstractSymEntry`. `GenSymEntry` is used to define the main metadata necessary for entry types that contain a single set of values and its necessary components. For example, a `SegArraySymEntry` stores three `SymEntries`, `values`, `segments`, and `lengths`, however these three pieces are all necessary to build a single instance of a `SegArray` object and reference a single set of data values.
+
+`SymEntry` is the entry type used for basic one-dimensional arrays of numeric values and their metadata. This is the most commonly used entry type.
+
+`SegArraySymEntry` tracks the component pieces of a `SegmentedArray` object. These components are all instances of `SymEntry` types.
+
+`SegStringSymEntry`tracks the component pieces of a `SegmentedStrings` object. The values of a `SegStringSymEntry` are stored in a `uint8` `SymEntry`. These components are all instances of `SymEntry` types.
+
+`CompositeSymEntry` is used to indicate that child `Entries` inheriting from this entry type contain multiple `data-adding` objects. This means that types under `CompositeSymEntry` contain multiple objects of varying types, which may not all be needed to conceptualize the Object. For example, `GroupBySymEntry` contains five `SymEntry` entries. One of these, `keyNamesEntry` could contain `n` number of references to additional data values. These could be of multiple types, and it is possible to build a GroupBy object with all or only part of the stored values.

```

Comment:
`This means that types under CompositeSymEntry contain multiple objects of varying types`

I get what this is saying but I feel like it could be confusing since types has two meanings here. Maybe we could reword it to something like

`This means that entry types under CompositeSymEntry contain multiple objects of varying types`

Also I feel like `Object` doesn't need to be capitalized here

Reply:
Updated